src/ast.rs

```diff
@@ -18,12 +18,21 @@ pub enum Expression<'a> {
     ParenExpression(Box<Expression<'a>>),
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq)]
 pub struct LocalAssignment<'a> {
     pub name: &'a str,
     pub value: Expression<'a>,
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct NumericFor<'a> {
+    pub var: &'a str,
+    pub start: Expression<'a>,
+    pub end: Expression<'a>,
+    pub step: Expression<'a>,
```

Comment:
Should `step` instead be an `Option<Expression<'a>>`, to indicate that the source might have not specified the step at all?

Reply:
It should, yes! Fixed now.